docs/design/schedule_algorithm.md

```diff
@@ -0,0 +1,39 @@
+
+- `NoDiskConflict`: Check if the free disk space meets the requirement of the Pod.
+- `PodFitsResources`: Check if the free resource (CPU and Memory) meets the requirement of the Pod. The free resource is measured by the capacity minus the sum of limits of all Pods on the node.
+- `PodFitsPorts`: Check if the port required by the Pod is already occupied on the node.
+- `PodFitsHost`: Filter out all nodes that the node name does not meet the Pod's requirement (you can set the Pod's preferred node name in its template).
+- `PodSelectorMatches`: Check if the labels of the node match the labels specified in the Pod's `nodeSelector` field.
+- `CheckNodeLabelPresence`: Check if all the specified labels exist on a node or not, regardless of the value. 
+- `CheckServiceAffinity`: Check if a node has specified labels for the purpose of service affinity. 
+
+The details of the above predicates can be found in [plugin/pkg/scheduler/algorithm/predicates/predicates.go](../../plugin/pkg/scheduler/algorithm/predicates/predicates.go). All predicates mentioned above can be used in combination to perform a sophisticated filtering policy. Kubernetes uses some of these predicates in the default scheduler and you can check the details in [plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go](../../plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go).
+
+## Ranking the nodes
+
+The filtered nodes are considered fitable to host the Pod, and it is often that there are more than one nodes remaining. Kubernetes prioritizes the remaining nodes to to find the "best" one for the Pod. The prioritization is performed by a set of priority functions. For each remaining node, a priority function gives a score which scales from 0-10 with 10 representing for "most preferred" and 0 for "least preferred". More over, each priority function is weighted by a positive number and the final score of each node is calculated by adding up all the weighted scores. For example, suppose there are two priority functions, `priorityFunc1` and `priorityFunc2` with weighting factors `weight1` and `weight2` respectively, the final score of some NodeA is:
+
+    finalScoreNodeA = (weight1 * priorityFunc1) + (weight2 * priorityFunc2)
+    
+After the scores of all nodes are calculated, the node with highest score is chosen as the host of the Pod. If there are more than one nodes with equal highest scores, a random one among them is chosen.
+
+Currently, Kuberntes scheduler provides some practical priority functions, including:
+
+- `LeastRequestedPriority`: The node is prioritized based on how much free resource (CPU and Memory) is available on it. The free resource is measured by the capacity minus the sum of the limits of all Pods already on the node, divided by the capacity.
```

Comment:
This isn't quite right. I would say "The node is prioritized based on the fraction of the node that would be free if the new Pod were scheduled onto the node. (In other words, (capacity - sum of limits of all Pods already on the node - limit of Pod that is being scheduled) / capacity). CPU and memory are equally weighted. The Pod with the highest fraction is the most preferred."